cl/corpus_importer/tasks.py

```diff
@@ -1599,10 +1672,30 @@ def update_rd_metadata(
     :param document_number: The docket entry number for use in file names.
     :param attachment_number: The attachment number (if applicable) for use in
     file names.
+    :param appellate: Whether the document belongs to an appellate court
+    obtained from recap.email
     :return: A two-tuple of a boolean indicating success and a corresponding
     error/success message string.
     """
+
     rd = RECAPDocument.objects.get(pk=rd_pk)
+
+    # Appellate documents obtained from recap.email don't contain a document
+    # number try to get it either from the PDF document or the download
+    # confirmation page.
+    if appellate:
```

Comment:
It's probably worth being defensive here. Perhaps:

```suggestion
    if appellate and not rd.document_number:
```

That will save you a microservice call or a PACER download for any doc that already has a value, right?

Reply:
Due to the new refactor, this condition is now within [`process_recap_email `](https://github.com/freelawproject/courtlistener/blob/2145-add-nda-support-for-recap-email/cl/recap/tasks.py#L1642)

So now it's not needed to look for the rd.document_number since docket entries for NDAs don't have a document number from the email report.